### PR TITLE
chg: redundant members list and organisations page #1013

### PR DESCRIPTION
**Pull request by cristianbell**
## Generic requirements in order to contribute to MISP:
- One Pull Request per fix/feature/change/...
- Keep the amount of commits per PR as small as possible: if for any reason, you need to fix your commit after the pull request, please squash the changes in one single commit (or tell us why not)
- Always make sure it is mergeable in the default branch (as of today 2016-06-03: branch 2.4)
- Please make sure Travis CI works on this request, or update the test cases if needed
- Any major changes adding a functionality should be disabled by default in the config
#### What does it do?

It resolves `#<1013>`
#### Questions
- [ ] Does it require a DB change?
- [ ] Are you using it in production?
- [ ] Does it require a change in the API (PyMISP for example)?
#### Release Type:
- [ ] Major
- [X] Minor
- [ ] Patch

### CODE REVIEW

File: app/View/Users/memberslist.ctp [hunk 1]
```diff
@@ -1,4 +1,5 @@
-<div class="users index">
+<div class="users index"><?php
+/*
```
Comment (rotanid) [inline app/View/Users/memberslist.ctp]: please just remove the code if it isn't necessary anymore, we can always take it from the git history if we need it again :)

**Comment by iglocska** [comment]
A few issues after a very brief check:
1. What Andreas pointed out already
2. The code in places if very hard to read, please use proper spacing
3. The code currently removes the members list but doesn't offer an alternative for users to have a quick overview over the organisations and the member count.
4. The PR removes the view portion of the members list but not the underlying look-ups. (in the Controller )
5. Please use the alternate if/for notation for code that envelops HTML

**Comment by cristianbell** [comment]
updated the core, hope it's all ok now.

**Comment by rotanid** [comment]
could you please fix the wrong indention of some lines in:
OrganisationsController.php
UsersController.php

File: app/Controller/UsersController.php [hunk 2]
```diff
@@ -726,18 +726,8 @@ public function resetauthkey($id = null) {
...
-				'fields' => $fields,
-				'recursive' => -1,
-				'contain' => array('Organisation'),
-				'group' => array('Organisation.name', 'Organisation.id'),
-				'order' => array('UPPER(Organisation.name)'),
-		);
-		$orgs = $this->User->find('all', $params);
-		$this->set('orgs', $orgs);
+	public function attributehistogram() {
+	    //all code is called via JS
```
Comment (Rafiot) [inline app/Controller/UsersController.php:730]: Why having the function then? (if it is a PHP weirdness, I apologize)

Comment (cristianbell) [inline app/Controller/UsersController.php:730]: it's needed to generate the view. If you call a /attributehistogram in the URL but there is no function with the same name in the controller the page will not be displayed. 
MVC principle + CakePHP. (and actually other fw work the same way here)

File: app/View/Users/attributehistogram.ctp [hunk 3]
```diff
@@ -0,0 +1,14 @@
+<div class="users index">
+	<div id = "histogram"></div>
+	<?php //echo $this->element('histogram');?>
```
Comment (rotanid) [inline app/View/Users/attributehistogram.ctp:3]: please remove this if it isn't used (as it's masked as a comment at the moment)

**Comment by rotanid** [comment]
assuming you have tested it's working, i think the code is ok.
todo:
- remove commented codeline
- rebase to current misp 2.4 branch state
- rebase this into one commit only (http://stackoverflow.com/a/5189600) and git push -f update the branch of the PR
  perhaps wait with the latter until @iglocska also checked and gave his OK so you don't have to do it multiple times :)

**Comment by iglocska** [comment]
I've accepted the commit and made some changes. Please review them for future reference. Some of it is more nice to have stuff (such as not breaking the layout with a long column name for a column that is going to be pretty small, indentation, etc) but some of it is crucial:
- correct use of contain
- not keeping obviously model level functions in the controller
- using the correct model when doing a lookup
- dealing with array indices if they are not set

Let me know if you have any questions about any of these.

**Pull request merged**